### PR TITLE
(Legacy/Segwit)AddressTest: add tests to show anomalies

### PR DESCRIPTION
**Pull request by msgilligan**
See the comments in PR #2613

### CODE REVIEW

File: core/src/test/java/org/bitcoinj/core/LegacyAddressTest.java [hunk 1]
```diff
@@ -77,6 +80,19 @@ public void decoding() {
...
+    public void equalityAnomaly() {
+        LegacyAddress a = LegacyAddress.fromBase58(TESTNET, "n4eA2nbYqErp7H6jebchxAN59DmNpksexv");
+        LegacyAddress b = LegacyAddress.fromBase58(SigNetParams.get(), "n4eA2nbYqErp7H6jebchxAN59DmNpksexv");
+        LegacyAddress c = LegacyAddress.fromBase58(RegTestParams.get(), "n4eA2nbYqErp7H6jebchxAN59DmNpksexv");
+        assertNotEquals(a, b);  // TODO: Passes, but probably shouldn't
+        assertNotEquals(b, c);  // TODO: Passes, but probably shouldn't
+        assertNotEquals(a, c);  // TODO: Passes, but probably shouldn't
+        assertEquals(a.toString(), b.toString());
+        assertEquals(b.toString(), c.toString());
+        assertEquals(a.toString(), c.toString());
```
Comment (msgilligan) [inline core/src/test/java/org/bitcoinj/core/LegacyAddressTest.java:93]: PR #2613 should fix this.

File: core/src/test/java/org/bitcoinj/core/SegwitAddressTest.java [hunk 2]
```diff
@@ -90,6 +93,44 @@ public void example_p2wpkh_testnet() {
...
     }
 
+    @Test
+    public void example_p2wpkh_testnet_equality_anomaly() {
+        String bech32 = "tb1qw508d6qejxtdg4y5r3zarvary0c5xw7kxpjzsx";
+
+        SegwitAddress a = SegwitAddress.fromBech32(TESTNET, bech32);
+        SegwitAddress b = SegwitAddress.fromBech32(SigNetParams.get(), bech32);
+
+        assertNotEquals(a, b);  // TODO: Passes, but probably shouldn't
```
Comment (msgilligan) [inline core/src/test/java/org/bitcoinj/core/SegwitAddressTest.java:103]: PR #2613 should fix this.

File: core/src/test/java/org/bitcoinj/core/SegwitAddressTest.java [hunk 3]
```diff
@@ -90,6 +93,44 @@ public void example_p2wpkh_testnet() {
...
+        assertEquals(ScriptType.P2WPKH, address.getOutputScriptType());
+        assertEquals(bech32.toLowerCase(Locale.ROOT), address.toBech32());
+        assertEquals(bech32.toLowerCase(Locale.ROOT), address.toString());
+    }
+
+    @Test
+    public void example_p2wpkh_regtest_fails() {
+        String bcrt1_bech32 = "bcrt1qspfueag7fvty7m8htuzare3xs898zvh30fttu2";
+
+        // TODO: Passes, but probably shouldn't
```
Comment (msgilligan) [inline core/src/test/java/org/bitcoinj/core/SegwitAddressTest.java:125]: A rethinking of how `Networks` is implemented (as part of the migration from `NetworkParameters` to `Network`) could fix this.

**Review by msgilligan — COMMENTED** [review]
I added comments on the anomalous tests and how they might be fixed.

File: core/src/test/java/org/bitcoinj/core/SegwitAddressTest.java [hunk 4]
```diff
@@ -90,6 +93,44 @@ public void example_p2wpkh_testnet() {
...
+        SegwitAddress a = SegwitAddress.fromBech32(TESTNET, bech32);
+        SegwitAddress b = SegwitAddress.fromBech32(SigNetParams.get(), bech32);
+
+        assertNotEquals(a, b);  // TODO: Passes, but probably shouldn't
+        assertEquals(a.toString(), b.toString());
+    }
+
+    @Test
+    public void example_p2wpkh_regtest() {
+        String bech32 = "bcrt1qspfueag7fvty7m8htuzare3xs898zvh30fttu2";
```
Comment (schildbach) [inline core/src/test/java/org/bitcoinj/core/SegwitAddressTest.java:109]: Perhaps rename this to `bcrt1_bech32` too (like below)?

**Comment by schildbach** [comment]
Maybe close this too, as #2613 is merged?

**Comment by msgilligan** [comment]
Closing in favor of PR #2613 (which contains the "equality equivalency" tests and is already merged) and PR #2631 which contains tests to parse Segwit REGTEST addresses.

**Pull request closed without merging**